build-tools/xaprepare/xaprepare/Steps/Step_InstallDotNetPreview.cs

```diff
@@ -74,18 +74,19 @@ async Task<bool> DownloadDotNetInstallScript (Context context, string dotnetScri
 			string tempDotnetScriptPath = dotnetScriptPath + "-tmp";
 			Utilities.DeleteFile (tempDotnetScriptPath);
 
-			Log.StatusLine ("Downloading dotnet-install...");
+			Log.StatusLine ("Downloading dotnet-install script...");
 
 			(bool success, ulong size, HttpStatusCode status) = await Utilities.GetDownloadSizeWithStatus (dotnetScriptUrl);
 			if (!success) {
 				string message;
 				if (status == HttpStatusCode.NotFound) {
-					message = "dotnet-install URL not found";
+					message = $"dotnet-install URL '{dotnetScriptUrl}' not found.";
 				} else {
-					message = $"Failed to obtain dotnet-install size. HTTP status code: {status} ({(int)status})";
+					message = $"Failed to obtain dotnet-install script size from URL '{dotnetScriptUrl}'. HTTP status code: {status} ({(int)status})";
 				}
 
-				return ReportAndCheckCached (message, quietOnError: true);
+				if (ReportAndCheckCached (message, quietOnError: true))
+					return true;
```

Comment:
I deleted the `quietOnError` parameter entirely and just made it log every time.

I don't think you'd see `message` above in the build logs without doing something like:

![image](https://github.com/xamarin/xamarin-android/assets/840039/cada22de-0d5d-4c74-a18a-3a4adff594f0)

Reply:
It seems like we may want to skip the cache file error logging initially, as long as we actually proceed to the download step which we weren't doing before. It could be confusing to ~always have that error message in our CI runs?

Reply:
The way it is currently, it looks like we'd never see this log message on error:

    Failed to obtain dotnet-install script size from URL '{dotnetScriptUrl}'. HTTP status code:

Reply:
Or we could simplify this to just warn in both places, as an error about failing to download the script will happen later either way...

Reply:
> The way it is currently, it looks like we'd never see this log message on error:
> 
> ```
> Failed to obtain dotnet-install script size from URL '{dotnetScriptUrl}'. HTTP status code:
> ```

Ah right, while this shouldn't be fatal, we still wouldn't know that the size check failed. Let me update the warning/error messaging again.